src/CustomerProfile.js

```diff
@@ -0,0 +1,30 @@
+import React, { useEffect, useState } from "react";
+
+function CustomerProfile({ id }) {
+  const [customerInfo, setCustomerInfo] = useState({});
+  useEffect(() => {
+    if (!id == 0) {
```

Comment:
It looks to me like condition `!id == 0` this will always evalute to true when `id` is truthy. Not sure what this condition is checking - remember that React will only trigger this effect if the `id` in the dependency array on line 11 updates.